### ISSUE TITLE
Matplotlib v3.3 deprecated some parameters in figure.colorbar and subsequent bug in proplot.

### ISSUE THREAD

**Issue by lacstorm** [lukelbd/proplot#232]
```python
import matplotlib.pyplot as plt
import proplot as pplt
import numpy as np
import matplotlib.colors as colors
import matplotlib.ticker as ticker


state = np.random.RandomState(51423)
data = (state.rand(20, 20))*50
data[5:15,5:15] = 0.002
data[10:20,10:20] = 0.6
#print(data.max(),data.min())

# define colormap
cmorph = [[255.,255,255],
   [0,225,225],
   [0,165,185],
   [0,185, 0],
   [0 ,255 , 45],
 [255, 255 ,  0],
 [225 ,205 ,  0],
 [160, 120 ,  0],
 [160 , 40 ,  0],
 [185 ,  0 ,  0],
 [255  , 0 ,155],
 [185  , 0 ,255],
 [125 ,  0, 255]]

cmorph = np.array(cmorph) / 255.0
cmap0 = colors.ListedColormap(cmorph[1:-1])
cmap0.set_under(cmorph[0])
cmap0.set_over(cmorph[-1])
levels = [0.1,0.5,1.,2.,3.,4.,5.,6.,8.,10.,20.,40]

# Figure with proplot
fig, ax = pplt.subplots()
m0 = ax.contourf(
  data, levels=levels, extend='both', cmap=cmap0,
  colorbar='b', colorbar_kw={'ticks':levels}
)

ax.format(suptitle='plot with proplot')
```

**When you use old version of matplotlib and run the code below, you'll get:**
![image](https://user-images.githubusercontent.com/35789946/98191141-8a6bcf80-1f53-11eb-8da7-5e0227dc5eb9.png)

**When you use new version of matplotlib (v3.3) and run the code below, you'll get a warning message:**

> D:\AnaConda\lib\site-packages\proplot\figure.py:1158: MatplotlibDeprecationWarning: The 'extend' parameter to Colorbar has no effect because it is overridden by the mappable; it is deprecated since 3.3 and will be removed two minor releases later.
  return super().colorbar(*args, cax=cax, **kwargs)

**And the figure is different:**
![image](https://user-images.githubusercontent.com/35789946/98191262-c4d56c80-1f53-11eb-942d-b11642fa20c6.png)

### Proplot version

0.6.4

**Comment by lacstorm** [comment]
It's solved.
We could just use boundary norm of matplotlib with the new parameter extend='both' in norm.

**Comment by lukelbd** [comment]
FYI this deprecation warning was recently fixed by a23e7043.
